shenyu-spring-boot-starter/shenyu-spring-boot-starter-client/shenyu-spring-boot-starter-client-springmvc/src/main/java/org/apache/shenyu/springboot/starter/client/springmvc/ShenyuSpringMvcClientInfoRegisterConfiguration.java

```diff
@@ -129,4 +133,23 @@ public ClientRegisterConfig clientRegisterConfig(final ShenyuClientConfig shenyu
                                                      final Environment env) {
         return new ClientRegisterConfigImpl(shenyuClientConfig, RpcTypeEnum.HTTP, applicationContext, env);
     }
+
+    /**
+     * InstanceRegisterListener.
+     *
+     * @param clientRegisterConfig  clientRegisterConfig
+     * @param shenyuDiscoveryConfig shenyuDiscoveryConfig
+     * @return InstanceRegisterListener
+     */
+    @Bean
+    @ConditionalOnBean(ShenyuDiscoveryConfig.class)
+    public InstanceRegisterListener instanceRegisterListener(final ClientRegisterConfig clientRegisterConfig, final ShenyuDiscoveryConfig shenyuDiscoveryConfig) {
+        DiscoveryUpstreamData discoveryUpstreamData = new DiscoveryUpstreamData();
+        discoveryUpstreamData.setUrl(clientRegisterConfig.getHost() + ":" + clientRegisterConfig.getPort());
+        discoveryUpstreamData.setStatus(0);
+        discoveryUpstreamData.setWeight(Integer.parseInt(shenyuDiscoveryConfig.getWeight()));
+        discoveryUpstreamData.setProtocol(ShenyuClientConstants.HTTP);
```

Comment:
protocol can come from the user's configuration?

Reply:
https?

Reply:
done